erdjs/src/smartcontracts/codec/enum.ts

```diff
@@ -0,0 +1,32 @@
+import { EnumType, U8Type, U8Value, EnumValue } from "../typesystem";
+import { BinaryCodec } from "./binary";
+
+export class EnumBinaryCodec {
+    private readonly parentCodec: BinaryCodec;
+
+    constructor(parentCodec: BinaryCodec) {
+        this.parentCodec = parentCodec;
+    }
+
+    decodeTopLevel(buffer: Buffer, type: EnumType): EnumValue {
+        let [decoded] = this.decodeNested(buffer, type);
```

Comment:
Function returns two values, but only one is used? Does it just take the first and ignore the second? (i.e. the EnumValue in this case?). Just making sure it's not a mistake.

Reply:
For `decodeTopLevel` we simply delegate to the `decodeNested` logic for enums, since they are equivalent (as far as I know, please confirm this). Indeed, we discard the `numRead` output parameter - which makes sense in the context of nested decode operations only - also please confirm this :).

Reply:
Simple enums (C-like) are serialized as an u8 either way, so yes, that is correct. Complex Rust-style enums remain to be discussed.

Reply:
Fixed, thanks!